sds/src/match_validation/http_validator.rs

```diff
@@ -136,71 +154,102 @@ impl HttpValidatorHelper {
 #[async_trait]
 impl MatchValidator for HttpValidator {
     async fn validate(&self, matches: &mut Vec<RuleMatch>, _: &[Box<dyn CompiledRuleDyn>]) {
-        // Let's reqwest the HTTP API endpoint to validate the matches
-        let futures = matches.iter_mut().map(|m| {
-            async move {
-                let mut request_builder: reqwest::RequestBuilder;
-                match self.config.method {
-                    HttpMethod::Get => {
-                        request_builder = HTTP_CLIENT.get(&self.config.endpoint);
-                    }
-                    HttpMethod::Post => {
-                        request_builder = HTTP_CLIENT.post(&self.config.endpoint);
-                    }
-                    HttpMethod::Put => {
-                        request_builder = HTTP_CLIENT.put(&self.config.endpoint);
-                    }
-                    HttpMethod::Delete => {
-                        request_builder = HTTP_CLIENT.delete(&self.config.endpoint);
-                    }
-                    HttpMethod::Patch => {
-                        request_builder = HTTP_CLIENT.patch(&self.config.endpoint);
-                    }
-                }
-                // Set timeout
-                request_builder = request_builder.timeout(self.config.options.timeout);
+        // build a map of match status per endpoint and per match_idx
+        let mut match_status_per_endpoint_and_match: AHashMap<_, _> = matches
+            .iter()
+            .enumerate()
+            .flat_map(|(idx, _)| {
+                self.config
+                    .endpoints
+                    .iter()
+                    .map(move |endpoint| ((idx, endpoint), MatchStatus::NotChecked))
+            })
+            .collect();
 
-                // Add headers
-                for header in &self.config.request_header {
-                    request_builder = request_builder.header(
-                        &header.key,
-                        &header.get_value_with_match(m.match_value.as_ref().unwrap()),
-                    );
-                }
-                let res = request_builder.send().await;
-                match res {
-                    Ok(val) => {
-                        // First check if this is in the valid status ranges
-                        for valid_range in &self.config.valid_http_status_code {
-                            if valid_range.contains(&val.status().as_u16()) {
-                                m.match_status = MatchStatus::Valid;
-                                return;
+        let futures = match_status_per_endpoint_and_match.iter_mut().map(
+            |((match_idx, endpoint), match_status)| {
+                let match_value = matches[*match_idx].match_value.as_ref().unwrap();
+                async move {
+                    let mut request_builder: reqwest::RequestBuilder;
+                    request_builder = match self.config.method {
```

Comment:
I don't think there is any conflict with `RequestBuilder` struct name, so there is not need to explicitly import it from  `reqwest`. Line 173 could be `let mut request_builder: RequestBuilder;`. Even better, you can assign directly the value to `request_builder`
```suggestion
                    let mut request_builder = match self.config.method {
```